pre_commit_hooks/mixed_line_ending.py

```diff
@@ -37,9 +37,15 @@ def fix_filename(filename, fix):
     # Some amount of mixed line endings
     mixed = sum(bool(x) for x in counts.values()) > 1
 
-    if fix == 'no' or (fix == 'auto' and not mixed):
+    if fix == 'no':
+        if mixed:
+            # Not fixing, just report problematic file
+            print('Mixed line endings: {filename}'.format(filename=filename))
         return mixed
 
+    if fix == 'auto' and not mixed:
+        return False
+
```

Comment:
a smaller patch which accomplishes the same thing and keeps the printing in `main()`:

```diff
-retv |= fix_filename(filename, args.fix)
+if fix_filename(filename, args.fix):
+    if args.fix == 'no':
+        print('{}: mixed line endings'.format(filename))
+    else:
+        print('{}: fixed mixed line endings.format(filename))
+    retv = 1
```